### PR TITLE
UF-234: Configurable Workbench Live Reload

### PR DESCRIPTION
**Pull request by paulovmr**
Depends on: https://github.com/uberfire/uberfire/pull/338

### CODE REVIEW

**Comment by paulovmr** [comment]
jenkins please retest this

**Comment by ederign** [comment]
@paulovmr we are ready to go?

**Comment by paulovmr** [comment]
@ederign Yes, we are.

File: uberfire-runtime-plugins/uberfire-runtime-plugins-backend/src/main/java/org/uberfire/ext/plugin/backend/PluginServicesImpl.java [hunk 1]
```diff
@@ -742,8 +802,11 @@ public FileVisitResult visitFile( final Path file,
         return result;
     }
 
+    IOService getIoService() {
```
Comment (ederign) [inline uberfire-runtime-plugins/uberfire-runtime-plugins-backend/src/main/java/org/uberfire/ext/plugin/backend/PluginServicesImpl.java]: @paulovmr next time, move this dependency to a construtor in order to mock it.

File: uberfire-runtime-plugins/uberfire-runtime-plugins-backend/src/main/java/org/uberfire/ext/plugin/backend/PluginServicesImpl.java [hunk 2]
```diff
@@ -525,31 +560,58 @@ public void delete( final org.uberfire.backend.vfs.Path path,
...
-                                                    getPluginContent( convert( newPath.resolve( path.getFileName() ) ) ),
+                                                    pluginContent,
                                                     sessionInfo ) );
 
         return result;
     }
 
+    private void removeRegistry( final Path path ) {
+        walkFileTree( path,
+                      new SimpleFileVisitor<Path>() {
```
Comment (ederign) [inline uberfire-runtime-plugins/uberfire-runtime-plugins-backend/src/main/java/org/uberfire/ext/plugin/backend/PluginServicesImpl.java]: we can use lambdas if is on master.

File: uberfire-runtime-plugins/uberfire-runtime-plugins-backend/src/main/java/org/uberfire/ext/plugin/backend/PluginServicesImpl.java [hunk 3]
```diff
@@ -337,44 +369,45 @@ private Path getDependencyPath( final Path pluginPath,
...
             final Path rootPlugin = getPluginPath( pluginName );
-            final DirectoryStream<Path> stream = ioService.newDirectoryStream( getCodeRoot( rootPlugin ),
-                                                                               new DirectoryStream.Filter<Path>() {
-                                                                                   @Override
-                                                                                   public boolean accept( final Path entry ) throws IOException {
-                                                                                       return entry.getFileName().toString().endsWith( ".code" );
-                                                                                   }
-                                                                               } );
+            final DirectoryStream<Path> stream = getIoService().newDirectoryStream( getCodeRoot( rootPlugin ),
+                                                                                    new DirectoryStream.Filter<Path>() {
```
Comment (ederign) [inline uberfire-runtime-plugins/uberfire-runtime-plugins-backend/src/main/java/org/uberfire/ext/plugin/backend/PluginServicesImpl.java]: we can use lambdas if is on master.

**Comment by ederign** [comment]
thanks @paulovmr f9c56fa723503140561d7ba4abb840d9766ffe95

**Pull request closed without merging**